frontend/components/commons/sidebar/TaskSidebar.vue

```diff
@@ -12,7 +12,12 @@
             :key="metric"
             v-if="currentMetric === metric"
           >
-            <component :is="componentName(metric)" :dataset="dataset" />
+            <component
+              :is="componentName(metric)"
+              :dataset="dataset"
+              :datasetId="dataset.id"
+              :datasetName="dataset.name"
```

Comment:
in some cases maybe you can use prop destructuring

Reply:
This refactoring is also part of the big refactoring to pass only necessary data [issue](https://github.com/argilla-io/argilla/issues/2142). So a better way to pass id and name is not by destructuring but by passing the corresponding props. It will be tackle in the corresponding refactoring issue.